libvcx/src/utils/devsetup_agent.rs

```diff
@@ -0,0 +1,462 @@
+        debug!("determine_message_type >>> a2a_message={:?}", a2a_message);
+        match a2a_message.clone() {
+            A2AMessage::PresentationRequest(_) => PayloadKinds::ProofRequest,
+            A2AMessage::CredentialOffer(_) => PayloadKinds::CredOffer,
+            A2AMessage::Credential(_) => PayloadKinds::Cred,
+            A2AMessage::Presentation(_) => PayloadKinds::Proof,
+            _msg => PayloadKinds::Other(String::from("aries"))
+        }
+    }
+
+    fn str_message_to_a2a_message(message: &str) -> VcxResult<A2AMessage> {
+        Ok(serde_json::from_str(message)
+            .map_err(|err| VcxError::from_msg(VcxErrorKind::InvalidJson, format!("Cannot deserialize A2A message: {}", err)))?
+        )
+    }
+
+    fn str_message_to_payload_type(message: &str) -> VcxResult<PayloadKinds> {
+        let a2a_message = str_message_to_a2a_message(message)?;
+        Ok(determine_message_type(a2a_message))
+    }
+
+    fn download_message(did: String, filter_msg_type: PayloadKinds) -> VcxAgencyMessage {
+        let mut messages = agency_client::get_message::download_messages_noauth(Some(vec![did]), Some(vec![String::from("MS-103")]), None).unwrap();
+        assert_eq!(1, messages.len());
+        let messages = messages.pop().unwrap();
+
+        for message in messages.msgs.into_iter() {
+            let decrypted_msg = &message.decrypted_msg.unwrap();
+            let msg_type = str_message_to_payload_type(decrypted_msg).unwrap();
+            if filter_msg_type == msg_type {
+                return VcxAgencyMessage {
+                    uid: message.uid,
+                    decrypted_msg: decrypted_msg.clone(),
+                };
+            }
+        }
+        panic!("Message not found")
+    }
+
+    pub trait TestAgent {
+        fn activate(&mut self) -> VcxResult<()>;
+    }
+
+    pub struct Faber {
+        pub is_active: bool,
+        pub config_wallet: WalletConfig,
+        pub config_agency: AgencyConfig,
+        pub config_issuer: IssuerConfig,
+        pub connection_handle: u32,
+        pub schema_handle: u32,
+        pub cred_def_handle: u32,
+        pub credential_handle: u32,
+        pub presentation_handle: u32,
+    }
+
+    impl TestAgent for Faber {
+        // propagte error up so we can handle it gracefully in
+        /// impl Drop for Faber
+        // so that we don't double panic, in case we are dropping due to a panick in the test
+        fn activate(&mut self) -> VcxResult<()> {
+            close_main_wallet()
+                .unwrap_or_else(|e| warn!("Failed to close main wallet (perhaps none was open?)"));
+            settings::clear_config();
+
+            info!("activate >>> Faber opening main wallet");
+            open_as_main_wallet(&self.config_wallet)?;
+            info!("activate >>> Faber initing issuer config");
+            init_issuer_config(&self.config_issuer)?;
+            info!("activate >>> Faber initing agency client");
```

Comment:
"initiating" (or "initting", if we are to make up words :)), and for Alice below as well

Reply:
fixed